manifests/server_common.pp

```diff
@@ -0,0 +1,62 @@
+# @summary Configure common settings to all OpenAFS server types
+#
+# @param files
+#   Hash of common server config files for OpenAFS servers
+#
+# @example
+#   include openafs::server_common
+class openafs::server_common (
+  Hash   $files,
+) {
+
+  File {
+    owner  => root,
+    group  => root,
+    ensure => file,
+    mode   => '0640',
+  }
+
+  ensure_resources('file', $files )
+
+  $puppet_file_header = '# This file is managed by Puppet; changes may be overwritten'
+
+  # SPECIFIC FILES FROM LOOKUP
+  $thiscell = lookup('openafs::thiscell')
```

Comment:
Suggest make these files as MODULE/files or MODULE/templates. Allow override contents via hiera parameter(s).

Reply:
At this point I prefer to be able to specify files and contents directly via hiera.